### PR TITLE
chore: improve codegen test coverage report

### PR DESCRIPTION
**Pull request by charles-cooper**
### What I did
go through coverage report of vyper/codegen, fix up bunch of little issues with `pragma: nocover`, branch coverage, delete some apparently dead code.

### How to verify it
check coverage report

### Commit message

Commit message for the final, squashed PR. (Optional, but reviewers will appreciate it! Please see [our commit message style guide](../../master/docs/style-guide.rst#best-practices-1) for what we would ideally like to see in a commit message.)

### Cute Animal Picture

Put a link to a cute animal picture inside the parenthesis-->

### CODE REVIEW

File: vyper/codegen/ir_node.py [hunk 1]
```diff
@@ -80,49 +80,10 @@ def resolve(self, body):
         yield mb, scoped_ir_nodes
 
 
-# create multiple with scopes if any of the items are complex, to force
-# ordering of side effects.
-@contextlib.contextmanager
-def scope_together(ir_nodes, names):
```
Comment (fubuloubu) [inline vyper/codegen/ir_node.py:86]: is this no longer used?

File: vyper/codegen/stmt.py [hunk 2]
```diff
@@ -414,9 +414,6 @@ def _is_terminated(code):
 
 # codegen a list of statements
 def parse_body(code, context, ensure_terminated=False):
-    if not isinstance(code, list):
```
Comment (fubuloubu) [inline vyper/codegen/stmt.py:417]: what validation shows this is not used anymore?

File: vyper/codegen/function_definitions/common.py [hunk 3]
```diff
@@ -51,7 +51,7 @@ def ir_identifier(self) -> str:
 
     def set_frame_info(self, frame_info: FrameInfo) -> None:
         # XXX: when can this happen?
-        if self.frame_info is not None:
+        if self.frame_info is not None:  # pragma: nocover
```
Comment (fubuloubu) [inline vyper/codegen/function_definitions/common.py:54]: Is this intended to be a compiler panic?

File: vyper/codegen/stmt.py (hunk 2 above)
Comment (charles-cooper) [reply vyper/codegen/stmt.py:417]: it's not in the coverage report. you can also check all the calls to parse_body and verify that they pass lists

File: vyper/codegen/ir_node.py (hunk 1 above)
Comment (charles-cooper) [reply vyper/codegen/ir_node.py:86]: that's right -- you can grep to verify

File: vyper/codegen/function_definitions/common.py (hunk 3 above)
Comment (charles-cooper) [reply vyper/codegen/function_definitions/common.py:54]: no -- i'm not sure why i added the pragma here

**Review by fubuloubu — APPROVED** [review]

**Pull request merged**